Add utility to obfuscate data in X12 files

kputnam opened the pull request:
This might help users who want to share examples from their production system, but need to scrub any sensitive data. This replaces all string fields (AN), date fields (DT), and time fields (TM) with dummy data, but should keep enough intact that the obfuscated file can still be parsed.

```
$ ruby -Ilib notes/obfuscate.rb spec/fixtures/X224-HC837/1-good.txt
ISA*00*__________*01*__________*ZZ*_______________*ZZ*_______________*151230*0000*^*00501*000000905*1*T*:~
GS*HC*__*__*20151230*0000*1*X*005010X224A1~
ST*837*____*005010X224A1~
BHT*0019*00*_*20151230*0000*CH~
NM1*41*2*_*****46*__~
PER*IC*_*TE*_~
NM1*40*2*_*****46*__~
HL*_**20*1~
NM1*85*2*_*****XX*__~
N3*_~
N4*__*FL*331111234~
REF*EI*_~
HL*_*_*22*1~
SBR*P********CI~
NM1*IL*1*_*_****MI*__~
N3*_~
N4*__*FL*33111~
NM1*PR*2*_*****PI*__~
N4*__*FL*33111~
HL*_*_*23*0~
PAT*19~
NM1*QC*1*_*_~
N3*_~
N4*__*FL*33413~
...
```

irobayna commented:
Good idea, this could be very handy.

Any reason why we would not put this under `bin` (like edi-pp) do is always available?

kputnam commented:
No reason, good idea!

kputnam commented:
I renamed it and added some "documentation" on how to use it, along with a flag to skip over unrecognized segments or elements (eg, when a `SegmentDef` has two elements but the input contained three).
